### PR TITLE
Add sa.sy.bluerecorder

### PR DESCRIPTION
**Pull request by me00001**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
    `--talk-name=org.gnome.Shell.Screencast` to record screen on GNOME Wayland 
    `--talk-name=org.kde.StatusNotifierWatcher` for tray icon
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 
    **Link:** 
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
    **Developer owns the domain**
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by me00001** [comment]
bot, build sa.sy.bluerecorder

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: any reason to exclude `aarch64` ?

File: sa.sy.bluerecorder.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
+app-id: sa.sy.bluerecorder
+runtime: org.gnome.Platform
+runtime-version: "3.38"
```
Comment (hfiguiere) [inline sa.sy.bluerecorder.yaml:3]: Please use runtime version 40

File: sa.sy.bluerecorder.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        CARGO_HOME: "/run/build/bluerecorder/cargo"
+        RUST_BACKTRACE: "1"
+    build-commands:
+      - mkdir -p $FLATPAK_DEST/opt $FLATPAK_DEST/share/icons/hicolor/128x128/apps/
+      - install -D flatpak/sa.sy.bluerecorder.metainfo.xml -t $FLATPAK_DEST/usr/share/metainfo/
+      - install -D flatpak/sa.sy.bluerecorder.desktop -t $FLATPAK_DEST/share/applications/
+      - cp -v data/blue-recorder@x128.png $FLATPAK_DEST/share/icons/hicolor/128x128/apps/sa.sy.bluerecorder.png
+      - cargo --offline fetch --manifest-path Cargo.toml --verbose 
+      - cargo --offline build --release --verbose
+      - cp target/release/blue-recorder $FLATPAK_DEST/opt
```
Comment (hfiguiere) [inline sa.sy.bluerecorder.yaml:39]: this should go in `$FLATPAK_DEST/bin` since it is the executable.

File: sa.sy.bluerecorder.yaml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  - org.freedesktop.Sdk.Extension.rust-stable
+cleanup:
+  - /share/man
+  - /share/doc
+  - /share/gtk-doc
+command: /app/opt/blue-recorder
+finish-args:
+  - --filesystem=home
+  - --share=ipc
+  - --socket=x11
```
Comment (hfiguiere) [inline sa.sy.bluerecorder.yaml:15]: if you have wayland support, this should be `--socket=fallback-x11`

File: sa.sy.bluerecorder.yaml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  - name: bluerecorder
+    buildsystem: simple
+    build-options:
+      append-path: /usr/lib/sdk/rust-stable/bin
+      env:
+        CARGO_HOME: "/run/build/bluerecorder/cargo"
+        RUST_BACKTRACE: "1"
+    build-commands:
+      - mkdir -p $FLATPAK_DEST/opt $FLATPAK_DEST/share/icons/hicolor/128x128/apps/
+      - install -D flatpak/sa.sy.bluerecorder.metainfo.xml -t $FLATPAK_DEST/usr/share/metainfo/
```
Comment (hfiguiere) [inline sa.sy.bluerecorder.yaml:34]: it's the wrong location, it should be `$FLATPAK_DEST/share/metainfo/`. This is likely why the buid failed

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: flathub.json (hunk 1 above)
Comment (me00001) [reply flathub.json:2]: I will add it, but i don't have `aarch64` to test it

File: sa.sy.bluerecorder.yaml (hunk 2 above)
Comment (me00001) [reply sa.sy.bluerecorder.yaml:3]: Ok

File: sa.sy.bluerecorder.yaml (hunk 3 above)
Comment (me00001) [reply sa.sy.bluerecorder.yaml:39]: Some directory must be placed in the same where executable is. as i know /opt is fine for that

File: sa.sy.bluerecorder.yaml (hunk 5 above)
Comment (me00001) [reply sa.sy.bluerecorder.yaml:34]: ok

**Comment by me00001** [comment]
bot, build sa.sy.bluerecorder

File: sa.sy.bluerecorder.yaml (hunk 4 above)
Comment (me00001) [reply sa.sy.bluerecorder.yaml]: done

File: flathub.json [hunk 6]
```diff
@@ -1,3 +1,6 @@
 {
```
Comment (hfiguiere) [inline flathub.json:1]: you can simply remove the flathub.json

Comment (me00001) [reply flathub.json:1]: I will do

**Comment by me00001** [comment]
bot, build sa.sy.bluerecorder

**Comment by xlmnxp** [comment]
> `flatpak install --user https://dl.flathub.org/build-repo/44704/sa.sy.bluerecorder.flatpakref`

It won't launch
```
$ flatpak run sa.sy.bluerecorder
Failed to initialize GTK.
```

**Comment by hfiguiere** [comment]
Well, looking at the code the app doesn't support wayland.
```rust
    // use "GDK_BACKEND=x11" to make xwininfo work in Wayland by using XWayland
    std::env::set_var("GDK_BACKEND", "x11");
    if gtk::init().is_err() {
        println!("Failed to initialize GTK.");
        return;
    }
```

So the permission should be changed.

**Comment by xlmnxp** [comment]
> Well, looking at the code the app doesn't support wayland.
> 
> ```rust
>     // use "GDK_BACKEND=x11" to make xwininfo work in Wayland by using XWayland
>     std::env::set_var("GDK_BACKEND", "x11");
>     if gtk::init().is_err() {
>         println!("Failed to initialize GTK.");
>         return;
>     }
> ```
> 
> So the permission should be changed.

it supported and work correctly without any issue, I'm the developer
when I build it locally by `flatpak-builder` it work just fine

**Comment by hfiguiere** [comment]
The code I quoted basically tell Gtk to use X11. So I stand by what I said. I you force the X11 backend, you don't support Wayland.
So the socket permissions should be solely X11.

**Comment by xlmnxp** [comment]
> The code I quoted basically tell Gtk to use X11. So I stand by what I said. I you force the X11 backend, you don't support Wayland.
> So the socket permissions should be solely X11.

it only for xwininfo to make it work in wayland
maybe I will replace xwininfo in future

**Comment by hfiguiere** [comment]
This code tell Gtk to use GdkX11. Past that whether wayland is here or not is irrelevant. The justifiaction doesn't matter (you are upstream, it's your call) but the result is that this is an X11 application. So the permission should be for an X11 only application. No wayland and `--socket=x11`

**Comment by me00001** [comment]
bot, build sa.sy.bluerecorder

**Comment by me00001** [comment]
Tell me if it's starting now @xlmnxp

**Comment by xlmnxp** [comment]
> Tell me if it's starting now @xlmnxp

it work now, can you downgrade it to
```yaml
runtime-version: "3.38"
```

**Comment by me00001** [comment]
> > Tell me if it's starting now @xlmnxp
> 
> it work now.
> 
Good :+1:

**Pull request closed without merging**